extlinks/links/models.py

```diff
@@ -45,6 +49,10 @@ class Meta:
         on_delete=models.SET_NULL,
         related_name="url",
     )
```

Comment:
bike shedding on code reuse for the hashtags tool:

If we wanted to move the link_events relationship here in preparation for making the events model reusable, we could define a field here with a one to many instead of a many to one (foreign key) in LinkEvents
https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.Field.one_to_many

That's not particularly nice though, IMO, so maybe we could just create an mediawiki-events app that provides an abstract model class and a bunch of helpers for dealing with the server side events and provide status pages.

Reply:
As far as I can tell, after looking at that section in the docs, it looks like those are attributes on an instance of a field rather than a field type. I don't see any way to define a field as a one to many relationship outside of declaring a foreign key which makes sense from a relational db perspective. In that case you would store the foreign key on the many(Linkevents) not the one (URLPatttern) row.  A list of all available field options is [here](Attributes for fields with relations[¶](https://docs.djangoproject.com/en/5.1/ref/models/fields/#attributes-for-fields-with-relations)).

Reply:
This would be used in a custom field class. But I may be overcomplicating things. A generic relationship might suffice. I'll try to whip up a quick example.